domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorkManagerInitializer.kt

```diff
@@ -11,45 +10,45 @@ import androidx.work.WorkManager
 import org.oppia.android.domain.oppialogger.ApplicationStartupListener
 import org.oppia.android.util.platformparameter.PlatformParameterValue
 import org.oppia.android.util.platformparameter.SyncUpWorkerTimePeriod
-import java.util.UUID
 import java.util.concurrent.TimeUnit
 import javax.inject.Inject
-import javax.inject.Singleton
 
 /**
  * Enqueues unique periodic work requests for fetching and caching latest platform parameter values
  * from the remote service on application creation.
  */
-@Singleton
 class PlatformParameterSyncUpWorkManagerInitializer @Inject constructor(
   private val context: Context,
   @SyncUpWorkerTimePeriod private val workRequestRepeatInterval: PlatformParameterValue<Int>
 ) : ApplicationStartupListener {
 
   private val OPPIA_PLATFORM_PARAMETER_WORK_REQUEST_NAME = "OPPIA_PLATFORM_PARAMETER_WORK_REQUEST"
 
-  private val platformParameterSyncUpWorkerConstraints = Constraints.Builder()
+  /** [Constraints] for platform parameter sync up work request. */
+  val platformParameterSyncUpWorkerConstraints = Constraints.Builder()
```

Comment:
Here & below: are these immutable structures? If not, we need to be really careful about exposing them like this.

Reply:
No they are not. Changed the approach now. PTAL